### ISSUE TITLE
RM2 support?

### ISSUE THREAD

**Issue by cweekly** [bordaigorl/remy#4]
This tool looks pretty cool. Does it support reMarkable 2?

**Comment by bordaigorl** [comment]
It should work in principle, but I cannot test it first hand.
The best performance in my tests is obtained using the `rsync` method which requires installing `rsync` on the tablet.
For that, you need to compile `rsync` for the correct version of the OS/chipset shipped with RM2.

**Comment by cweekly** [comment]
Thanks! Much appreciated.

**Comment by mefistoso** [comment]
When I try to run `python remygui.py wifi` (wifi is rsync), I'm getting the following error complaining it doesn't find a `templates.json`file:

```python
File "/home/user/Downloads/software/remarkable/remy/remy/remarkable/filesource.py", line 372, in __init__
    with open(self._local("templates.json", branch=TEMPLDIR), 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: '~/remarkable/rsync/templates/templates.json'
```

Not sure what this `templates.json` file is supposed to be or where to get one to adapt. FWIW my `remy.json` config file is at https://gist.github.com/mefistoso/c2b2ab081ecf1949a4af3fe3765a412b

**Comment by bordaigorl** [comment]
Hi @mefistoso I can help you with that but would you mind opening a new issue for this? Your question is not related to RM2 support...

**Comment by karlicoss** [comment]
I checked `local` method with Remarkable 2 (have a separate SSH backup script), and it works. The only couple of issues
- one notebook out of few isn't opening and the app gets unresponsive when I try to.. I'll try to debug this
- pdfs aren't opening, but it's more likely to be some poppler/QT issue

**Comment by bordaigorl** [comment]
@karlicoss thanks so much for testing it!
Let me know if you find bugs.
Unfortunately reMy is still quite unfinished, I am planning to work on it but I have very limited time.
Which OS are you on? I could provide binaries to see if the poppler/QT issue is resolved.
(I also added new instruction in the README on how to install the stubborn python-poppler-pyqt5)

**Comment by karlicoss** [comment]
@bordaigorl  No problem! Thanks for your time spent implementing this :)
I've moved in a separate issue, since it seems irrelevant to RM2 support: https://github.com/bordaigorl/remy/issues/7

**Comment by bordaigorl** [comment]
@karlicoss regarding unresponsiveness: a common reason for inefficient rendering is the heavy use of eraser.
I know it sounds stupid, but the eraser is really annoying/inefficient to render correctly. This is why reMy offers various methods to render it. If you set the eraser mode to `ignore` for previews you'd probably get those notebooks to display normally (although some details of the rendering might be off). See the Readme for details.
To my knowledge, none of the other exporters render the eraser accurately when exporting to vectorial form (most would be equivalent to reMy's `ignore` mode).

If you have any idea on how to make that more efficient I'm all ears!

**Comment by karlicoss** [comment]
Thanks, indeed, setting it to `ignore` helped! 
I guess in theory it's possible to detect when the file takes too long to open and suggest this setting to the user. Or maybe, always open in `ignore` mode first, and only after that try to apply the more accurate setting (but this might get too complicated). 

I didn't quite understand from the readme (and from a brief look at pagerender.py), what's the core cause of slow erasing? Is this because without eraser drawing is relatively straightforward -- you need to draw the layers from bottom to top; whereas the eraser acts on all the layers simultaneously? Or is it because it's not a 'native' operation in QT canvas, and you have to 'subtract' the strokes?

**Comment by bordaigorl** [comment]
@karlicoss I answered your question in #8

**Comment by benneti** [comment]
I think this can be closed as RM2 works fine
